### PR TITLE
protocol: client: enable builtin yamux client support

### PR DESCRIPTION
**Pull request by bergwolf**
When enableYamux is set, dial the grpc server with builtin yamux
translation support.

Fixes: #200

### CODE REVIEW

File: protocols/client/client_test.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func TestNewAgentClient(t *testing.T) {
 	assert.Nil(t, err, "failed to start mock server: %s", err)
 
 	cliFunc := func(sock string, success bool, expect string) {
-		cli, err := NewAgentClient(sock)
+		cli, err := NewAgentClient(sock, false)
```
Comment (jodh-intel) [inline protocols/client/client_test.go]: Is there no way to unit test the yamux option?

Comment (bergwolf) [reply protocols/client/client_test.go]: surely there is. I'll add UT for it.

Comment (jodh-intel) [reply protocols/client/client_test.go]: Thanks @bergwolf ! :smile:

**Comment by jodh-intel** [comment]
Thanks @bergwolf - `TestNewAgentClientWithYamux()` failed under Travis though (test timed out).

**Comment by bergwolf** [comment]
@jodh-intel CI test timeout is too low. Can we increase it?
`panic: test timed out after 10s`

The UT actually includes two `context deadline timeout` tests that only return after waiting for 5 seconds each.

**Comment by jodh-intel** [comment]
Hi @bergwolf - yes, we *could* change the timeout...

- https://github.com/kata-containers/tests/blob/master/.ci/go-test.sh#L14

... but 10 seconds is a long time. How about doing this instead?:

```diff
diff --git a/protocols/client/client.go b/protocols/client/client.go
index ffaded3..b092524 100644
--- a/protocols/client/client.go
+++ b/protocols/client/client.go
@@ -26,9 +26,11 @@ import (
 const (
 	unixSocketScheme  = "unix"
 	vsockSocketScheme = "vsock"
-	dialTimeout       = 5 * time.Second
 )
 
+// defined as a var to allow the tests to modify it
+var dialTimeout = 5 * time.Second
+
 // AgentClient is an agent gRPC client connection wrapper for agentgrpc.AgentServiceClient
 type AgentClient struct {
 	agentgrpc.AgentServiceClient
diff --git a/protocols/client/client_test.go b/protocols/client/client_test.go
index b7789b3..633da23 100644
--- a/protocols/client/client_test.go
+++ b/protocols/client/client_test.go
@@ -12,6 +12,7 @@ import (
 	"os"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/yamux"
 	"github.com/stretchr/testify/assert"
@@ -106,6 +107,13 @@ func agentClientTest(t *testing.T, sock string, success, enableYamux bool, expec
 }
 
 func TestNewAgentClient(t *testing.T) {
+	savedDialTimeout := dialTimeout
+	dialTimeout = 1 * time.Second
+
+	defer func() {
+		dialTimeout = savedDialTimeout
+	}()
+
 	mock, waitCh, err := startMockServer(t, false)
 	assert.Nil(t, err, "failed to start mock server: %s", err)
 
@@ -129,6 +137,13 @@ func TestNewAgentClient(t *testing.T) {
 }
 
 func TestNewAgentClientWithYamux(t *testing.T) {
+	savedDialTimeout := dialTimeout
+	dialTimeout = 1 * time.Second
+
+	defer func() {
+		dialTimeout = savedDialTimeout
+	}()
+
 	mock, waitCh, err := startMockServer(t, true)
 	assert.Nil(t, err, "failed to start mock server: %s", err)
```

**Comment by bergwolf** [comment]
@jodh-intel thanks for the suggestion. I've added it. PTAL.

**Comment by jodh-intel** [comment]
Hi @bergwolf - Nice - much neater your way! ;)

lgtm

**Pull request merged**